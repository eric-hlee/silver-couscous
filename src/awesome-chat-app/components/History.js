import React from 'react'
import SetActiveChat from '../containers/SetActiveChat'


class History extends React.Component {

  render() {

    const style_history = {
      height: '600px',
      overflow: 'auto',
    }

    return (
      <div className="History" style={style_history}>
        <div className="ChatList">
          {this.props.chatList.map(chat =>
            <SetActiveChat
              key={chat.id}
              id={chat.id}
              lastMessage={chat.lastMessage}
              onClick={this.props.onClick}
            />
          )}
        </div>
      </div>
    );
  }
}

export default History