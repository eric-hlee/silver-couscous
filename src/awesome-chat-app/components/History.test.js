import React from 'react'
import { shallow } from 'enzyme'
import History from './History'

describe('History', () => {
  const mockChatList = [
    {
      id: 0,
      messageList: [
        {
          id: 0,
          time: '2019-06-07T09:28:01.467Z',
          message: 'Welcome to the chat',
          fromUser: false
        }
      ]
    }
  ]
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<History chatList={mockChatList}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
