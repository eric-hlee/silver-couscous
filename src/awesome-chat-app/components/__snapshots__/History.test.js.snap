// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should render self and subcomponents 1`] = `
<div
  className="History"
  style={
    Object {
      "height": "600px",
      "overflow": "auto",
    }
  }
>
  <div
    className="ChatList"
  >
    <Connect(ChatCard)
      id={0}
      key="0"
    />
  </div>
</div>
`;
